packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -0,0 +1,198 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' as convert;
+import 'dart:io' as io;
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:platform/platform.dart';
+
+import 'package:flutter_goldens_client/client.dart';
+
+// If you are here trying to figure out how to use golden files in the Flutter
+// repo itself, consider reading this wiki page:
+// https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package%3Aflutter
+
+// TODO(Piinks): This file will replace ./client.dart when transition to Skia
+// Gold testing is complete
+const String _kFlutterRootKey = 'FLUTTER_ROOT';
+const String _kGoldctlKey = 'GOLDCTL';
+const String _kServiceAccountKey = 'GOLD_SERVICE_ACCOUNT';
+
+/// A class that represents the Skia Gold client for golden file testing.
+class SkiaGoldClient {
+  /// Create a handle to a local workspace for the Skia Gold Client
+  SkiaGoldClient({
+    this.fs = const LocalFileSystem(),
+    Platform platform = const LocalPlatform(),
+  }) : _platform = platform;
+
+  /// The file system to use for storing local files for running image tests.
+  ///
+  /// This is useful in tests, where a local file system (the default) can be
+  /// replaced by a memory file system.
+  final FileSystem fs;
+
+  /// A wrapper for the [dart:io.Platform] API.
+  ///
+  /// This is only for use in tests, where the system platform (the default) can
+  /// be replaced by mock platform instance.
+  Platform get platform => _platform;
+
+  Platform _platform;
+
+  /// The local [Directory] where the Flutter repository is hosted.
+  ///
+  /// Uses the [fs] file system.
+  Directory get flutterRoot => fs.directory(platform.environment[_kFlutterRootKey]);
+
+  /// The local [Directory] root of all image tests.
+  ///
+  /// Uses the [fs] file system.
+  Directory get comparisonRoot => flutterRoot.childDirectory(fs.path.join('bin', 'cache', 'pkg', 'goldens'));
```

Comment:
Why does SkiaGold need to know about this? I though that's the checkout of the goldens repository that Skia Gold is supposed to replace?